front_end/src/components/landing_page/search_widget/SearchWidget.js

```diff
@@ -78,6 +82,27 @@ class SearchWidget extends Component {
       this.props.handleResetSearchOverview();
     }
 
+    //Check if the destination is given or not.
+    if (this.state.destinationName === "") {
+      this.setState({ showDesAlert: true });
+    } else {
+      this.setState({ showDesAlert: false });
+    }
+    //Check if the checkIn and checkOut date are given or not.
+    if (this.state.checkIn === "" || this.state.checkOut === "") {
+      this.setState({ showDateAlert: true });
+    } else {
+      this.setState({ showDateAlert: false });
+    }
+
+    if (
+      this.state.destinationName !== "" &&
+      this.state.checkIn !== "" &&
+      this.state.checkOut !== ""
+    ) {
+      this.props.history.push("/searchResultOverview");
+    }
```

Comment:
saveQuery and submitQuery should also be within the if statement, otherwise we call the backend with incomplete data.